### PR TITLE
Refactor X/Y/Z values to uint

### LINKED ISSUE #312: Move z,x,y values to from int to uint
Z/X/Y values should never be negative and therefor should be of type `uint`. This change will need to be implemented in the `server` package for both `HandleMapLayerZXY` and `HandleMapZXY` methods, the `cache` interface and sub packages as well as other packages that accept any z/x/y values (i.e. atlas package: `Map.FilerLayersByZoom()`)

### PR DESCRIPTION
**Pull request by ear7h**
This should close #312 and #334

<s><b>todo</b>
* tile struct in root tile.go</s>

### CODE REVIEW

File: server/handle_map_layer_zxy_test.go [hunk 1]
```diff
@@ -73,6 +87,7 @@ func TestHandleMapLayerZXY(t *testing.T) {
 
 		r, err := http.NewRequest(test.reqMethod, test.uri, nil)
 		if err != nil {
+
```
Comment (gdey) [inline server/handle_map_layer_zxy_test.go]: Let's remove this extra newline.

**Comment by ARolek** [comment]
The v0.6.0 branch no longer exists. moving this to be based of the new v0.7.0 branch.

**Comment by gdey** [comment]
The changes that are needed here is the rename of the import paths.

File: cache/redis/redis.go [hunk 2]
```diff
@@ -71,24 +73,29 @@ func New(config map[string]interface{}) (rcache cache.Interface, err error) {
 		return nil, fmt.Errorf("redis did not respond with 'PONG', '%s'", pong)
 	}
 
+	// the config map's underlying value is int
```
Comment (ARolek) [inline cache/redis/redis.go]: If we change this to look for a uint we can drop the less than 0 check on line 82

File: cache/s3/s3.go [hunk 3]
```diff
@@ -62,17 +64,19 @@ func New(config map[string]interface{}) (cache.Interface, error) {
 	//	parse the config
 	c := dict.M(config)
 
-	//	TODO: this could be cleaner
-	defaultMaxZoom := 0
+	// the config map's underlying value is int
+	defaultMaxZoom := tegola.MaxZ
 	maxZoom, err := c.Int(ConfigKeyMaxZoom, &defaultMaxZoom)
```
Comment (ARolek) [inline cache/s3/s3.go]: We can change this to parse for a Uint and then drop the less than 0 check on line 74

**Comment by ARolek** [comment]
This commit looks generally fine to me, but we still have a bit of inconsistency in our `uint` strategy. We have a combination of `uint` and `uint64` sprinkled around the codebase all in relation to z/x/y values. We should pick a single way to represent those values and stick with it. I go back and forth on which one to use. `uint` would be nice as it will choose the optimal bit size to use based on the architecture. `uint64` is nice as that's what `strconv` returns when parsing values so it saves on an extra line of code here and there.

I don't have a specific preference on which type we use, we just need to be consistent throughout the codebase when representing Z/X/Y values.

**Comment by gdey** [comment]
I say we just go with uint

File: cache/s3/s3.go (hunk 3 above)
Comment (ear7h) [reply cache/s3/s3.go]: Ah, missed those

**Comment by ARolek** [comment]
LGTM

**Pull request merged**